### PR TITLE
refactored strings from master to control plane

### PR DESCRIPTION
**Pull request by gooney152**
#### What type of PR is this?
/kind cleanup
/wg-naming

#### What this PR does / why we need it:
updates the new standard of referring to master as control plane

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
NONE

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/framework/test_context.go [hunk 1]
```diff
@@ -342,28 +342,28 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
...
+	flags.StringVar(&cloudConfig.MasterName, "kube-control-plane", "", "Name of the kubernetes control plane. Only required if provider is gce or gke")
 	flags.StringVar(&cloudConfig.APIEndpoint, "gce-api-endpoint", "", "The GCE APIEndpoint being used, if applicable")
 	flags.StringVar(&cloudConfig.ProjectID, "gce-project", "", "The GCE project being used, if applicable")
 	flags.StringVar(&cloudConfig.Zone, "gce-zone", "", "GCE zone being used, if applicable")
 	flags.Var(cliflag.NewStringSlice(&cloudConfig.Zones), "gce-zones", "The set of zones to use in a multi-zone test instead of querying the cloud provider.")
 	flags.StringVar(&cloudConfig.Region, "gce-region", "", "GCE region being used, if applicable")
 	flags.BoolVar(&cloudConfig.MultiZone, "gce-multizone", false, "If true, start GCE cloud provider with multizone support.")
-	flags.BoolVar(&cloudConfig.MultiMaster, "gce-multimaster", false, "If true, the underlying GCE/GKE cluster is assumed to be multi-master.")
+	flags.BoolVar(&cloudConfig.MultiMaster, "gce-multicontrolplane", false, "If true, the underlying GCE/GKE cluster is assumed to be multi-control-plane.")
```
Comment (wangyysde) [inline test/e2e/framework/test_context.go:359]: Should "control-plane" be conformance to above and below?

Comment (gooney152) [reply test/e2e/framework/test_context.go:359]: @MorrisLaw

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -342,28 +342,28 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.Tooling, "tooling", "", "The tooling in use (kops, gke, etc.)")
 	flags.StringVar(&TestContext.OutputDir, "e2e-output-dir", "/tmp", "Output directory for interesting/useful test data, like performance data, benchmarks, and other metrics.")
 	flags.StringVar(&TestContext.Prefix, "prefix", "e2e", "A prefix to be added to cloud resources created during testing.")
-	flags.StringVar(&TestContext.MasterOSDistro, "master-os-distro", "debian", "The OS distribution of cluster master (debian, ubuntu, gci, coreos, or custom).")
+	flags.StringVar(&TestContext.MasterOSDistro, "control-plane-os-distro", "debian", "The OS distribution of cluster control plane (debian, ubuntu, gci, coreos, or custom).")
```
Comment (BenTheElder) [inline test/e2e/framework/test_context.go:345]: If you rename a public facing API like a command line flag we also need to have a plan for updating callers setting that flag and it probably needs a release note for callers we do not control.

Comment (BenTheElder) [reply test/e2e/framework/test_context.go:345]: Renaming these makes sense, but we need to take some consideration in how we go about enacting it to not just break CI. One approach would be roughly:
- add two flags for some time
- get CI migrated
- have a release note about the old name being deprecated
- having it print a warning if the old name is used
- in some future release deleting the old flag

breaking changes with this many steps usually have a KEP so we've documented the plan somewhere. https://github.com/kubernetes/kubernetes/issues/103144#issuecomment-876826747

Comment (BenTheElder) [reply test/e2e/framework/test_context.go:345]: I think it probably makes the most sense to leave flags/environment variables to a different PR so the other more easily merged changes can merge quicker while someone scopes out the external / breaking changes

**Comment by dims** [comment]
Is this PR still needed, please rebase if so (or we can close it?)

**Pull request closed without merging**